misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/CustomSystemClassLoader.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.ballerinalang.test.runtime;
+
+import io.ballerina.projects.util.ProjectConstants;
+
+import java.io.*;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+public class CustomSystemClassLoader extends ClassLoader {
+
+    private final HashMap<String, byte[]> modifiedClassDefs;
+    private final List<String> excludingClasses = new ArrayList<>();
+
+    public CustomSystemClassLoader() {
+        super(getSystemClassLoader());
+        this.modifiedClassDefs = new HashMap<>();
+
+        populateExcludingClasses();
+    }
+
+    private void populateExcludingClasses() {
+        try(InputStream is = BTestMain.class.getResourceAsStream("/" + ProjectConstants.EXCLUDING_CLASSES_FILE);
+            BufferedReader br = new BufferedReader(new InputStreamReader(Objects.requireNonNull(is)))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                excludingClasses.add(line);
+            }
+        }
+        catch (NullPointerException | IOException e) {
+            throw new RuntimeException("Error reading excludingClasses.txt", e);
+        }
+
```

Comment:
```suggestion
```

Reply:
Done in [40229a4](https://github.com/ballerina-platform/ballerina-lang/pull/42183/commits/40229a40c6e8385fc2d56f004746b7a5494411f1) commit.